goal_src/levels/jungleb/plat-flip.gc

```diff
@@ -5,3 +5,419 @@
+      (set! (-> s5-5 quad) (-> self base-pos quad))
+      (+! (-> s5-5 y) (* 1638.4 (update! (-> self smush))))
+      (TODO-RENAME-30 (-> self root-override) s5-5)
+      )
+     (suspend)
+     )
+    )
+   (none)
+   )
+  :post
+  (the-as (function none :behavior plat-flip) rider-post)
+  )
+
+;; definition for method 11 of type plat-flip
+;; Used lq/sq
+(defmethod copy-defaults! plat-flip ((obj plat-flip) (arg0 res-lump))
+  (local-vars
+   (r0-0 uint128)
+   (a0-17 uint128)
+   (a0-26 uint128)
+   (a1-11 uint128)
+   (sv-16 int)
+   (sv-32 int)
+   (sv-48 int)
+   )
+  (set! (-> obj mask) (logior (process-mask platform) (-> obj mask)))
+  (let
+   ((s4-0
+     (new 'process 'collide-shape-moving obj (collide-list-enum hit-by-player))
+     )
+    )
+   (set! (-> s4-0 dynam) (copy *standard-dynamics* 'process))
+   (set! (-> s4-0 reaction) default-collision-reaction)
+   (set! (-> s4-0 no-reaction) nothing)
+   (dummy-29 s4-0 1)
+   (let
+    ((s3-0
+      (new
+       'process
+       'collide-shape-prim-mesh
+       s4-0
+       (the-as uint 0)
+       (the-as uint 0)
+       )
+      )
+     )
+    (set! (-> s3-0 prim-core collide-as) (the-as uint 2048))
+    (set! (-> s3-0 collide-with) (the-as uint 16))
+    (set! (-> s3-0 prim-core action) (the-as uint 3))
+    (set! (-> s3-0 prim-core offense) 4)
+    (set! (-> s3-0 transform-index) 3)
+    (set-vector! (-> s3-0 local-sphere) 0.0 0.0 0.0 11468.8)
+    )
+   (dummy-46 s4-0)
+   (set! (-> s4-0 nav-radius) (* 0.75 (-> s4-0 root-prim local-sphere w)))
+   (dummy-50 s4-0)
+   (set! (-> obj root-override) s4-0)
+   )
+  (process-drawable-from-entity! obj arg0)
+  (set! (-> obj base-pos quad) (-> obj root-override trans quad))
+  (dummy-14 obj *plat-flip-sg* '())
+  (logior! (-> obj skel status) 1)
+  (let ((f30-0 300.0))
+   (set! sv-16 0)
+   (let
+    ((v1-28
+      (res-lump-data
+       arg0
+       'delay
+       pointer
+       :tag-ptr
+       (the-as (pointer res-tag) (& sv-16))
+       )
+      )
+     )
+    (set! (-> obj before-turn-down-time) (* f30-0 (if (and v1-28 (begin
+                                                                  (let
+                                                                   ((a0-16
+                                                                     (the-as
+                                                                      uint128
+                                                                      sv-16
+                                                                      )
+                                                                     )
+                                                                    )
+                                                                   (.pcpyud
+                                                                    a0-17
+                                                                    a0-16
+                                                                    r0-0
```

Comment:
this use of r0 won't work properly, but I think this should go away when sv-16 is a res-tag